presto-tests/src/test/java/com/facebook/presto/tests/TestQueryManager.java

```diff
@@ -138,7 +142,10 @@ public void testFailQueryPrerun()
                         "SELECT * FROM lineitem")
                 .get();
 
-        assertNotEquals(dispatchManager.getStats().getQueuedQueries(), 0L, "Expected 0 queued queries, found: " + dispatchManager.getStats().getQueuedQueries());
+        //Wait for query to be in queued state
+        while (dispatchManager.getStats().getQueuedQueries() != 1) {
```

Comment:
Should we also add a condition for `dispatchManager.getStats().getRunningQueries() != 0` just to make sure this doesn't end up in an infinite loop? This can happen in the case one or more queries finish execution before (this query is queued) or (this condition has detected that it is queued). Although, it is likely that this never happens just to have a safeguard in place.

Reply:
The test will timeout if this goes into infinite loop with the 60s timeout. So the test will fail in that case.

Reply:
Oh yes, thanks I missed the timeout.

Reply:
that is a good question @imjalpreet . We just talked about the same here as well. :)